log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+                    },
+                    "title": {
+                        "fill": {"value": config.legendTitleColor},
+                        "fontSize": {"value": config.legendTitleFontSize}
+                    },
+                    "labels": {
+                        "fill": {"value": config.legendTextColor},
+                        "fontSize": {"value": config.ledgendTextFontSize}
+                    }
+                }
+            }
+        ];
+
+        this.spec.legends = legends;
+    }
+
+
+    var axes =  getXYAxes(config, "x", "x", "y", "y");
+
+    marks.push(getAreaMark(config, this.metadata));
+    config.fillOpacity  = 0;
+    config.markSize = 1000;
+    marks.push(getSymbolMark(config, this.metadata));
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.axes = axes;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+};
+
+area.prototype.draw = function(div, callbacks) {
+
+    var viewUpdateFunction = (function(chart) {
+        if(this.config.tooltip.enabled){
+            createTooltip(div);
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+            bindTooltip(div,this.view,this.config,this.metadata);
+        } else {
+            this.view = chart({el:div}).renderer(this.config.renderer).update();
+        }
+
+        if (callbacks != null) {
+            for (var i = 0; i<callbacks.length; i++) {
+                this.view.on(callbacks[i].type, callbacks[i].callback);
+            }
+        }
+
+    }).bind(this);
+
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
+            var allowedDataSet = [];
+            var startingPoint = dataset.length - maxValue;
+            for(var i = startingPoint; i < dataset.length;i++){
+                allowedDataSet.push(dataset[i]);
+            }
+            this.spec.data[0].values = allowedDataSet;
+        }
+    }
+
+    vg.parse.spec(this.spec, viewUpdateFunction);
+};
+
+area.prototype.insert = function(data) {
+    //Removing events when max value is enabled
+    if (this.config.maxLength != -1 && this.config.maxLength <  (this.view.data(this.config.title).values().length + data.length)) {
+        var removeFunction = (function(d) {
+            return d[this.metadata.names[this.config.x]] == oldData;
+        }).bind(this);
+
+        for (i = 0; i < data.length; i++) {
+            var oldData = this.view.data(this.config.title).values()[i][this.metadata.names[this.config.x]];
+            this.view.data(this.config.title).remove(removeFunction);
+        }
+    }
+
+    this.view.data(this.config.title).insert(data);
+    this.view.update();
+};
+
+area.prototype.getSpec = function() {
+    return this.spec;
+};
+
+
+function getAreaMark(config, metadata){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD